### ISSUE TITLE
[BUG] version 3.1.4 both server/client bug, client on Windows 11

### ISSUE THREAD

**Issue by Kalimeiro** [grke/burp#926]
Hi,

We have a bug, not all times, on a Windows 11 Pro Client and VSS, see partial logs :

The server/client have latest build 3.1.4

The server is on Debian 12 BookWorm.

```
2024-02-05 11:59:58: SSL is using cipher: TLS_AES_256_GCM_SHA384  TLSv1.3 Kx=any      Au=any  Enc=AESGCM(256) Mac=AEAD
2024-02-05 11:59:58: extra_comms_begin ok:autoupgrade:incexc:orig_client:uname:failover:vss_restore:regex_icase:counters_json:msg:forceproto=1:rshash=blake2:seed:
2024-02-05 11:59:58: Compression level: 5
2024-02-05 11:59:58: do backup client
2024-02-05 11:59:58: Using librsync hash blake2
2024-02-05 11:59:58: Control handler registered.
2024-02-05 12:00:01: Generate VSS snapshots.
2024-02-05 12:00:01: Driver="VSS Vista", Drive(s)="C"
2024-02-05 12:00:01: WARNING: Generate VSS snapshots failed.ERR=L�op�ration a r�ussi.
2024-02-05 12:00:01: Problem with VSS
```

VSS snapshot failed with an error : operation successfull (in french).

**Comment by grke** [comment]
Hello,
OK, thank you, I think I need to add extra logging before that ERR= line.

**Comment by grke** [comment]
Extra logging added in git master.

**Comment by grke** [comment]
Hello,

You could build it yourself, or wait for my next release, or you can download the one that I just built here:

https://burp.grke.org/downloads/test/burp-win64-installer-3.1.5.exe

**Comment by ohmer1** [comment]
Hi,

I have the same issue on one specific Windows 10 PC. Both client and server are running Burp version 2.4.0 and I tried your 3.1.5 test build on the problematic client. Unfortunately, I don't see more verbose logs.

```
2024-07-09 08:58:01 -0400: burp[1577467] Backup started
2024-07-09 08:58:01 -0400: burp[1577467] Client version: 3.1.5
2024-07-09 08:58:01 -0400: burp[1577467] Protocol: 1
2024-07-09 08:58:01 -0400: burp[1577467] Using librsync hash blake2
2024-07-09 08:58:01 -0400: burp[1577467] Client is Windows
2024-07-09 08:58:01 -0400: burp[1577467] WARNING: Client 'client-name' version '3.1.5' does not match server version '2.4.0'. An upgrade is recommended.
2024-07-09 08:58:01 -0400: burp[1577467] Begin phase1 (file system scan)
2024-07-09 08:58:04 -0400: burp[1577467] WARNING: Generate VSS snapshots failed.ERR=L�op�ration a r�ussi.
2024-07-09 08:58:04 -0400: burp[1577467] got error: Problem with VSS

2024-07-09 08:58:13 -0400: burp[1577467] error in phase 1
2024-07-09 08:58:13 -0400: burp[1577467] Backup failed
```

Is it possible there is something wrong caused by the localization? I notice that both the original issue reporter and me seem to run Windows with french locales. The "error" is saying the operation succeeded, it doesn't make sense.

**Comment by ohmer1** [comment]
Log extract from client instead of from the server this time:

```
2024-07-09 13:28:23: C:\Program Files\Burp\bin\burp.exe[6872] do backup client
2024-07-09 13:28:23: C:\Program Files\Burp\bin\burp.exe[6872] Using librsync hash blake2
2024-07-09 13:28:23: C:\Program Files\Burp\bin\burp.exe[6872] Control handler registered.
2024-07-09 13:28:26: C:\Program Files\Burp\bin\burp.exe[6872] Generate VSS snapshots.
2024-07-09 13:28:26: C:\Program Files\Burp\bin\burp.exe[6872] Driver="VSS Vista", Drive(s)="C"
2024-07-09 13:28:26: C:\Program Files\Burp\bin\burp.exe[6872] Failed PrepareForBackup
2024-07-09 13:28:26: C:\Program Files\Burp\bin\burp.exe[6872] WARNING: Generate VSS snapshots failed.ERR=LÆopÚration a rÚussi.

2024-07-09 13:28:26: C:\Program Files\Burp\bin\burp.exe[6872] Problem with VSS
```

**Comment by ohmer1** [comment]
Sorry to spam, but something I found I forgot to report: rebooting the PC resolve the issue and Burp is able to backup again but only for short time before the issue come back again.

**Comment by grke** [comment]
Hello,
It sounds like there is something wrong with your PC.
Maybe try looking in the Windows event logs?

**Comment by promasu** [comment]
I'm able to use burp in that constellation reliably. 3.1.4 on Windows 11 Pro and Debian 12.

**Comment by ohmer1** [comment]
In my case, I found the problem is caused by our security product (Sentinel One). Adding the burp executable in the exclusion list is not enough. I found this workaround in the interoperability section of the S1 documentation. This workaround was written for another backup software but it appears to work fine with Burp too.

> Interoperability with Unitrends
> 
> Supported from Management version: Eiffel
> 
> Supported from Agent version Windows 2.8
> 
> This issue was fixed in Windows Agent 4.5.2.
> 
> What you see:
> 
> When SentinelOne and Unitrends are installed on the same endpoint, Unitrends backup does not work.
> 
> Our Development team is working to resolve this in an upcoming Windows release.
> 
> How to resolve:
> 
> How you resolve this depends on the version of the Windows Agent.
> 
> Note: This workaround does NOT prevent endpoints from taking Windows snapshots as usual.
> 
> Windows Agent versions 4.2 SP2 - 4.4
> 
> To resolve in Windows Agent versions 4.2 SP2 - 4.4, disable "vssConfig.agentVssWriters" through Sentinelctl or Policy Override.
> 
> Example in Sentinelctl:
> 
>  sentinelctl config -p agent.vssConfig.agentVssWriters -v false -k "passphrase"
> 
> Example in Policy Override:
> 
> { 	
>     "vssConfig": { 		
>         "agentVssWriters": false 	
>     } 
> }
> 
> Windows Agent version 4.5.1
> 
> To resolve in Windows Agent versions 4.5.1, disable "vssConfig.enableResearchDataCollectorVssWriter" through Sentinelctl or Policy Override.
> 
> Example in Sentinelctl:
> 
> sentinelctl config -p agent.vssConfig.enableResearchDataCollectorVssWriter -v false -k "passphrase"
> 
> Example in Policy Override:
> 
> { 	
>     "vssConfig": { 		
>         "enableResearchDataCollectorVssWriter": false 	
>     } 
> }

I opened a feature request to ask S1 to add support for Burp interoperability in the future. It has been logged as FSKEY-5533. Feel free to ask S1 support for this feature request too if you have this issue.

**Comment by grke** [comment]
Great, thank you for the information.

Given that this is not a problem with burp, I will close this issue.
